client/src/client.rs

```diff
@@ -1126,14 +1124,22 @@ impl RpcApi for Client {
         cmd: &str,
         args: &[serde_json::Value],
     ) -> Result<T> {
-        let req = self.client.build_request(&cmd, &args);
+        let raw_args: Vec<_> = args
+            .iter()
+            .map(|a| {
+                let json_string = serde_json::to_string(a)?;
```

Comment:
This is causing a lot of allocations, is it really necessary? What's the reason/benefit to serialize a JSON string there instead of only in the end? It seems to be 1.29 related, but why do we need raw values in the first place?

EDIT: now I see that has to do with the choices made in `jsonrpc`. It's probably not too bad in the grand scheme of things but also not great.

Reply:
Yes, ordinarily we'd just pull it into a raw_value. But this intermediate allocation is required for compatibility with old versions of serde_json which are rooted in the MSV of 1.29